python/ql/src/semmle/python/frameworks/Django.qll

```diff
@@ -1891,7 +1908,9 @@ private module Django {
   class DjangoRouteHandlerRequestParam extends django::http::request::HttpRequest::InstanceSource,
     RemoteFlowSource::Range, DataFlow::ParameterNode {
     DjangoRouteHandlerRequestParam() {
-      this.getParameter() = any(DjangoRouteSetup setup).getARouteHandler().getRequestParam()
+      this.getParameter() = any(DjangoRouteSetup setup).getARequestHandler().getRequestParam()
+      or
+      this.getParameter() = any(DjangoViewClassHandlerWithoutKnownRoute setup).getRequestParam()
```

Comment:
```suggestion
      // Handlers by route
      this.getParameter() = any(DjangoRouteSetup setup).getARequestHandler().getRequestParam()
      or
      // Handlers with no route
      this.getParameter() = any(DjangoViewClassHandlerWithoutKnownRoute setup).getRequestParam()
```

Reply:
I guess we are capturing (request parameters for) Django request handlers here, which is too specific (being more specific than just request handlers) to already be captured by a predicate. Perhaps we should name it explicitly?

Reply:
I don't fully understand your follow up comment, about naming this explicitly. I've added a QLDoc for the class definition (since it was missing) -- does this clear up things?

About adding the comments, do you feel strongly about that? I feel they are just repeating what the QL code say (but maybe I'm biased since I wrote the code in the first place).

Reply:
"Naming it explicitly" here means defining a predicate or class, say
```
class DjangoRequestHandler extends HTTP::Server::RequestHandler {
  DjangoRequestHandler() {
    this = any(DjangoRouteSetup setup).getARequestHandler()
    or
    this instanceof DjangoViewClassHandlerWithoutKnownRoute
  }
}
```
That would capture the fact that there are handlers of two kinds and likely need no comment. The above code could then be simplified from "there are parameters from two sources because there are handlers of two kinds" to `any(DjangoRequestHandler handler).getRequestParam()`.

All this just to explain my thinking. It might be overengineering and I do not actually feel strongly about it, nor about the comments (but hopefully this explains why I put them in).